Feature/fix query alteration options

edarroudi opened the pull request:
The query alteration structure is different on request and response.
See https://docs.microsoft.com/de-de/graph/search-concept-speller
The options have to be inside the "request" structure, not outside.
The queryAlterationResponse object is inside the "value" response.

I don't know if i/we did some mistake or Microsoft moved the structure inside the request. The updated article date suggests that something did change. Perhaps @wobba knows more about it. 
![image](https://user-images.githubusercontent.com/935493/152037831-dd33faab-f9f4-44fb-8c64-f5d535590570.png)

edarroudi commented:
Saw this interesting piece as well.. 
![image](https://user-images.githubusercontent.com/935493/152039402-6ac1724c-18f1-4e01-8be9-675513378d5f.png)

Shall i change how we handle the queryTemplate as well or is it too soon? Currently:
![image](https://user-images.githubusercontent.com/935493/152039661-d84d3bf8-b07a-4b6f-bc6a-789d78a16832.png)

FranckyC commented:
Hi @edarroudi , `queryAlterationResponse` is still only available in beta so we won't pocess it until it moves ot v1.0 endpoint. Thanks!

edarroudi commented:
Hi, it's not beta anymore: https://developer.microsoft.com/en-us/graph/changelog/?page=2
I modified it (no requirement for beta flag) and updated the documentation.
![image](https://user-images.githubusercontent.com/935493/156448617-1e7d6468-49f9-4e6a-89a0-57e7f65a1400.png)
